optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetFilterBiNode.java

```diff
@@ -43,6 +45,11 @@ public void addChildNode(BavetAbstractBiNode<A, B> childNode) {
         childNodeList.add(childNode);
     }
 
+    @Override
+    public List<BavetAbstractBiNode<A, B>> getChildNodes() {
+        return Collections.unmodifiableList(childNodeList);
```

Comment:
(important IMO)
The entire Bavet implementation is not public API. We don't need to wrap in `unmodifiableList()` etc, as that will only create perf loss and memory usage increase.